WalletWasabi/LocalNode.cs

```diff
@@ -0,0 +1,108 @@
+using System;
+using System.Linq;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using NBitcoin;
+using NBitcoin.Protocol;
+using NBitcoin.Protocol.Behaviors;
+using WalletWasabi.Backend.Models;
+using WalletWasabi.Helpers;
+using WalletWasabi.Logging;
+using WalletWasabi.Services;
+
+namespace WalletWasabi
+{
+	public class LocalNodeWrapper
+	{
+		private Node _node;
+		private WasabiSynchronizer _synchronizer;
+
+		public Node Internal => _node; 
+
+		public static SyncBehavior SyncWatcher { get; } = new SyncBehavior();
+
+		public EventHandler<EventArgs> FullySynchronized;
+
+		public bool IsFullySynchronized { get; private set; }
+
+		public LocalNodeWrapper(Node node)
+		{
+			_node = node;
+			SyncWatcher.Synchronized += async (s, e)=>
+				await CheckFullSynchronizationAsync();
+		}
+
+		public void Watch(WasabiSynchronizer synchronizer)
+		{
+			_synchronizer = synchronizer;
+			_synchronizer.ResponseArrived += async (s, args) => {
+				if( args.FiltersResponseState == FiltersResponseState.NoNewFilter )
+				{
+					await CheckFullSynchronizationAsync();
+				}
+				SyncWatcher.UpdateKnowTip(_synchronizer.BestKnownFilter.BlockHash);
+			};
+		}
+
+		private async Task CheckFullSynchronizationAsync()
+		{
+			try
+			{
+				await Task.Run(()=>{
+					var BlockHashes = new[] { _synchronizer.BestKnownFilter.BlockHash };
+					var blocks = _node.GetBlocks(BlockHashes);
+					if(blocks != null && blocks.Any(x=>x.GetHash() == BlockHashes[0]))
+					{
+						IsFullySynchronized = true;
+						var syncEvent = FullySynchronized;
+						if(syncEvent != null)
+							syncEvent(this, EventArgs.Empty);
+					}
+				});
+			}
+			catch(Exception)
+			{
+				// TODO: handle this
+			}
+		}
+
+		public static LocalNodeWrapper Connect(IPEndPoint endpoint, Network network, NodeConnectionParameters connectionParameters)
+		{
+			var handshakeTimeout = new CancellationTokenSource();
+			handshakeTimeout.CancelAfter(TimeSpan.FromSeconds(5));
+
+			var localhost = new NetworkAddress(endpoint);
+			var nodeConnectionParameters = connectionParameters.Clone();
+			nodeConnectionParameters.ConnectCancellation = handshakeTimeout.Token;
+			nodeConnectionParameters.IsRelay = false;
```

Comment:
This should be true, since now we want the mempool transactions, too.
https://github.com/zkSNACKs/WalletWasabi/pull/1216#issuecomment-470952979

Reply:
Ops, you're right, thx